pkg/logentry/stages/replace.go

```diff
@@ -0,0 +1,218 @@
+package stages
+
+import (
+	"bytes"
+	"reflect"
+	"regexp"
+	"text/template"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+)
+
+// Config Errors
+const (
+	ErrEmptyReplaceStageConfig = "empty replace stage configuration"
+	ErrEmptyReplaceStageSource = "empty source in replace stage"
+)
+
+// ReplaceConfig contains a regexStage configuration
+type ReplaceConfig struct {
+	Expression string  `mapstructure:"expression"`
+	Source     *string `mapstructure:"source"`
+	Replace    string  `mapstructure:"replace"`
+}
+
+// validateReplaceConfig validates the config and return a regex
+func validateReplaceConfig(c *ReplaceConfig) (*regexp.Regexp, error) {
+	if c == nil {
+		return nil, errors.New(ErrEmptyReplaceStageConfig)
+	}
+
+	if c.Expression == "" {
+		return nil, errors.New(ErrExpressionRequired)
+	}
+
+	if c.Source != nil && *c.Source == "" {
+		return nil, errors.New(ErrEmptyReplaceStageSource)
+	}
+
+	if c.Replace == "" {
+		return nil, errors.New(ErrEmptyReplaceStageConfig)
```

Comment:
also, I know this PR is merged, but why is this validated? If, for example, I want to remove part of the log, I cannot use replace pipeline as it won't accept an empty replace. I could use a whitespace, but then it looks funny

Reply:
I think it's a fair request. You can open a PR or may be @adityacs is up for it.

Reply:
👍

Reply:
I think I'll have some free time today or tomorrow to do a PR. Just wanted to ask your opinions about it :)